monitor.py

```diff
@@ -158,17 +159,30 @@ def run(self):
         if start == self.NEVER:
             start = None
 
-        cutoff = datetime.datetime.utcnow()
-        new_timestamp_value = self.run_once(start, cutoff) or cutoff
-        duration = datetime.datetime.utcnow() - cutoff
-        self.cleanup()
+        exception = None
+        started_running = datetime.datetime.utcnow()
+        try:
+            new_timestamp_value = (
+                self.run_once(start, started_running) or started_running
+            )
+            self.cleanup()
+        except Exception, e:
+            self.log.error("Error running %s monitor", exc_info=e)
+            exception = traceback.format_exc()
```

Comment:
It would be better to pass the exception itself around and get the stack trace later but I don't see an easy way to do that in Python 2.